### PR TITLE
Move reader-main to reader/components

### PR DESCRIPTION
**Pull request by bluefuton**
### Changes proposed in this Pull Request
A number of Reader-specific components and blocks currently live in the main components and blocks directories, even though they are not used elsewhere in Calypso. I feel that these components and blocks should live inside client/reader if they are not likely to be reused outside Reader. See also: #28161.

### What's reader-main anyway?

> This class is used by pieces of the Reader to indicate they want "editorial" styles. Notably, this overrides the background color of the document and is used as a hook by other parts to override styles.

### Testing instructions
Load the Reader at http://calypso.localhost:3000 and ensure that all Reader streams have a white background.

### CODE REVIEW

File: client/blocks/reader-full-post/index.jsx [hunk 1]
```diff
@@ -113,6 +113,10 @@ export class FullPostView extends React.Component {
 			this.attemptToSendPageView();
 		}
 
+		if ( this.props.shouldShowComments && ! prevProps.shouldShowComments ) {
```
Comment (bluefuton) [inline client/blocks/reader-full-post/index.jsx]: This is an unrelated change - the linter picked up on usage of `componentWillReceiveProps` in Reader full post. 

To test, make sure that a full post URL with `#comments` suffix still takes you to the comments section of a full post, e.g.:

http://calypso.localhost:3000/read/blogs/140522955/posts/2773#comments

File: assets/stylesheets/_components.scss [hunk 2]
```diff
@@ -169,7 +169,6 @@
 @import 'components/progress-bar/style';
 @import 'components/pulsing-dot/style';
 @import 'components/purchase-detail/style';
-@import 'components/reader-main/style';
```
Comment (jsnmoon) [inline assets/stylesheets/_components.scss]: What are the consequences of removing this?

**Review by jsnmoon — DISMISSED** [review]
Code changes look good and reader seems to work as expected. Needs a rebase to proceed, though.

File: assets/stylesheets/_components.scss (hunk 2 above)
Comment (bluefuton) [reply assets/stylesheets/_components.scss]: Yes! It's now loaded directly in the component:

https://github.com/Automattic/wp-calypso/pull/28272/files#diff-5e2375393afe7f203d69e6ca92bed475R17

This is part of the wider effort to migrate CSS build to Webpack - see https://github.com/Automattic/wp-calypso/pull/26820 and p4TIVU-98Q-p2.

Comment (jsnmoon) [reply assets/stylesheets/_components.scss]: Ah, right! Totally missed that one.

File: client/blocks/reader-full-post/index.jsx (hunk 1 above)
Comment (jsnmoon) [reply client/blocks/reader-full-post/index.jsx]: A suggestion for the future: Lazy loading images can push the scroll of the page further down, moving the comments section off-screen. I'd imagine this could be solved if we had the height information of the images.

**Review by jsnmoon — APPROVED** [review]
